<a href="https://colab.research.google.com/github/oimartin/460/blob/main/geriatric_visits_nb.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Injest

In [1]:
import pandas as pd

In [2]:
visits = pd.read_csv("https://raw.githubusercontent.com/oimartin/Older-and-Wiser/geriatric_visits/data/Geriatric_Medicine.csv")

# EDA

In [3]:
visits.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 42959 entries, 0 to 42958
Data columns (total 15 columns):
 #   Column                                                Non-Null Count  Dtype  
---  ------                                                --------------  -----  
 0   zip_code                                              42959 non-null  int64  
 1   min_medicare_pricing_for_new_patient                  42959 non-null  float64
 2   max_medicare_pricing_for_new_patient                  42959 non-null  float64
 3   mode_medicare_pricing_for_new_patient                 42959 non-null  float64
 4   min_copay_for_new_patient                             42959 non-null  float64
 5   max_copay_for_new_patient                             42959 non-null  float64
 6   mode_copay_for_new_patient                            42959 non-null  float64
 7   most_utilized_procedure_code_for_new_patient          42959 non-null  int64  
 8   min_medicare_pricing_for_established_patient          42

In [4]:
visits.iloc[:, 0] = visits['zip_code'].astype('string')
visits.iloc[:, 7] = visits.iloc[:, 7].astype('string')
visits.iloc[:, 14] = visits.iloc[:, 14].astype('string')

In [6]:
visits.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 42959 entries, 0 to 42958
Data columns (total 15 columns):
 #   Column                                                Non-Null Count  Dtype  
---  ------                                                --------------  -----  
 0   zip_code                                              42959 non-null  string 
 1   min_medicare_pricing_for_new_patient                  42959 non-null  float64
 2   max_medicare_pricing_for_new_patient                  42959 non-null  float64
 3   mode_medicare_pricing_for_new_patient                 42959 non-null  float64
 4   min_copay_for_new_patient                             42959 non-null  float64
 5   max_copay_for_new_patient                             42959 non-null  float64
 6   mode_copay_for_new_patient                            42959 non-null  float64
 7   most_utilized_procedure_code_for_new_patient          42959 non-null  string 
 8   min_medicare_pricing_for_established_patient          42

In [20]:
distrib_df = visits.describe()
distrib_df

min_medicare_pricing_for_new_patient  \
count                          42959.000000   
mean                              58.238339   
std                                4.045550   
min                               53.136000   
25%                               55.312000   
50%                               56.936000   
75%                               59.432000   
max                               74.816000   

       max_medicare_pricing_for_new_patient  \
count                          42959.000000   
mean                             177.239252   
std                               11.124559   
min                              163.672000   
25%                              169.736000   
50%                              174.056000   
75%                              180.408000   
max                              233.632000   

       mode_medicare_pricing_for_new_patient  min_copay_for_new_patient  \
count                           42959.000000               42959.000000   
mean                              177.239252                  14.559585   
std                                11.124559                   1.011387   
min                               163.672000                  13.284000   
25%                               169.736000                  13.828000   
50%                               174.056000                  14.234000   
75%                               180.408000                  14.858000   
max                               233.632000                  18.704000   

       max_copay_for_new_patient  mode_copay_for_new_patient  \
count               42959.000000                42959.000000   
mean                   44.309813                   44.309813   
std                     2.781140                    2.781140   
min                    40.918000                   40.918000   
25%                    42.434000                   42.434000   
50%                    43.514000                   43.514000   
75%                    45.102000                   45.102000   
max                    58.408000                   58.408000   

       min_medicare_pricing_for_established_patient  \
count                                  42959.000000   
mean                                      18.073701   
std                                        1.459078   
min                                       16.256000   
25%                                       17.040000   
50%                                       17.616000   
75%                                       18.544000   
max                                       23.768000   

       max_medicare_pricing_for_established_patient  \
count                                  42959.000000   
mean                                     144.793623   
std                                        9.072558   
min                                      133.848000   
25%                                      138.800000   
50%                                      142.080000   
75%                                      147.312000   
max                                      190.744000   

       mode_medicare_pricing_for_established_patient  \
count                                   42959.000000   
mean                                      103.643492   
std                                         6.628021   
min                                        95.608000   
25%                                        98.992000   
50%                                       101.496000   
75%                                       105.648000   
max                                       135.848000   

       min_copay_for_established_patient  max_copay_for_established_patient  \
count                       42959.000000                       42959.000000   
mean                            4.518425                          36.198406   
std                             0.364770                           2.268139   
min                             4.064000                          33.462000   
25%           

In [8]:
print("most_utilized_procedure_code_for_new_patient")
display(visits.iloc[:, 7].value_counts())
print("most_utilized_procedure_code_for_established_patient")
display(visits.iloc[:, 14].value_counts())

most_utilized_procedure_code_for_new_patient


99205    42959
Name: most_utilized_procedure_code_for_new_patient, dtype: Int64

most_utilized_procedure_code_for_established_patient


99214    42959
Name: most_utilized_procedure_code_for_established_patient, dtype: Int64

In [19]:
visits.nunique()

zip_code                                                42959
min_medicare_pricing_for_new_patient                       95
max_medicare_pricing_for_new_patient                       96
mode_medicare_pricing_for_new_patient                      96
min_copay_for_new_patient                                  95
max_copay_for_new_patient                                  96
mode_copay_for_new_patient                                 96
most_utilized_procedure_code_for_new_patient                1
min_medicare_pricing_for_established_patient               90
max_medicare_pricing_for_established_patient               96
mode_medicare_pricing_for_established_patient              96
min_copay_for_established_patient                          90
max_copay_for_established_patient                          96
mode_copay_for_established_patient                         96
most_utilized_procedure_code_for_established_patient        1
dtype: int64

99214: Established patient office visit, 30-39 minutes
<br>
99205: Typically 60 minutes, Comprehensive history, Comprehensive Examination, High complexity medical decision making

# Aggregate

In [ ]:
bins = [0, 1, 5, 10, 25, 50, 100]
visits['binned'] = pd.cut(visits['percentage'], bins)
print (df)